src/Dropdown/index.jsx

```diff
@@ -1,225 +1,198 @@
 
     return (
       <div
+        {...other}
         className={classNames(
           'dropdown',
+          className,
           {
-            show: open,
-            'has-icon': hasIconElement,
-            rounded: hasIconElement,
-            border: hasIconElement,
-            'd-flex': hasIconElement,
-            'bg-white': hasIconElement,
+            show: this.state.open,
           },
         )}
         ref={(container) => { this.container = container; }}
       >
-        { hasIconElement &&
-          <div
-            className={classNames(
-              'icon-container',
-              'd-flex',
-              'align-items-center',
-              'justify-content-center',
-              'border-right',
-            )}
-          >
-            {React.cloneElement(iconElement, {
-              className: iconElement.props ? classNames(iconElement.props.className, 'rounded-left') : null,
-            })}
-          </div>
-        }
         <Button
-          aria-expanded={open}
+          aria-expanded={this.state.open}
           aria-haspopup="true"
-          buttonType={buttonType}
+          className={buttonClassName}
           onClick={this.toggle}
-          onKeyDown={this.handleToggleKeyDown}
-          className={classNames(
-            'dropdown-toggle',
-            {
-              'border-0': hasIconElement,
-              'rounded-0': hasIconElement,
-              'bg-white': hasIconElement,
-            },
-          )}
           type="button"
-          inputRef={(toggleElem) => { this.toggleElem = toggleElem; }}
+          inputRef={(toggleButton) => { this.toggleButton = toggleButton; }}
         >
-          {title}
+          {buttonContent}
         </Button>
+        {/* eslint-disable-next-line jsx-a11y/interactive-supports-focus */}
         <div
-          aria-label={title}
-          aria-hidden={!open}
           className={classNames(
             'dropdown-menu',
-            { show: open },
+            className,
+            {
+              show: this.state.open,
+            },
           )}
+          aria-label={buttonContent}
+          aria-hidden={!this.state.open}
           role="menu"
+          ref={this.menuItems}
+          onKeyDown={this.handleMenuKeyDown}
         >
-          {this.generateMenuItems(menuItems)}
+          {this.props.children}
         </div>
       </div>
     );
   }
 }
 
 Dropdown.propTypes = {
-  buttonType: PropTypes.string,
-  iconElement: PropTypes.element,
-  menuItems: PropTypes.arrayOf(
-    PropTypes.shape({
-      label: PropTypes.string,
-      href: PropTypes.string,
-    }),
-    PropTypes.element,
-  ).isRequired,
-  title: PropTypes.string.isRequired,
+  className: PropTypes.string,
+  buttonClassName: PropTypes.string,
+  children: PropTypes.node.isRequired,
+  buttonContent: PropTypes.string.isRequired,
 };
 
 Dropdown.defaultProps = {
-  buttonType: 'light',
-  iconElement: undefined,
+  buttonClassName: 'dropdown-toggle',
```

Comment:
Is `dropdown-toggle` needed at all? If it is let's bake in 'dropdown-toggle' into our component instead of being the default so consumers of it don't need to know this class name when they add more.

Reply:
Also, let's set the default to `btn-light` instead

Reply:
Updated using [classnames](https://www.npmjs.com/package/classnames) `className={classNames(
            'dropdown-toggle',
            buttonClassName,
          )}`
and added 'btn-light' as default.